compiler/token/token.go

```diff
@@ -67,6 +67,9 @@ const (
 	If     = "IF"
 	ElsIf  = "ELSIF"
 	Else   = "ELSE"
+	Case   = "CASE"
+	When   = "WHEN"
+	Then   = "THEN"
```

Comment:
@shuheiktgw When will we have to use the `then` keyword? I think it's redundant since most Rubyist ignore this keyword. Any other thoughts? @goby-lang/maintainers

Reply:
I thought I think about porting ruby gems into Goby ecosystem, and if so, I think we need this... Otherwise I agree `then` keyword is redundant!  👍

Reply:
@shuheiktgw Um....I think we should remove `then` when porting those gems instead of supporting it. And I don't see many people use this keyword nowadays actually, do you see a lot?